<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1569</width>
    <height>1046</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QStackedWidget" name="stackedWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>1551</width>
      <height>1011</height>
     </rect>
    </property>
    <widget class="QWidget" name="pageText">
     <widget class="QPushButton" name="btnBack">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>850</y>
        <width>80</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Back</string>
      </property>
     </widget>
     <widget class="QTextEdit" name="textInput">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>10</y>
        <width>381</width>
        <height>91</height>
       </rect>
      </property>
     </widget>
     <widget class="QTextEdit" name="textOutput">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>150</y>
        <width>381</width>
        <height>131</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="btnCompressText">
      <property name="geometry">
       <rect>
        <x>319</x>
        <y>110</y>
        <width>91</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Compress</string>
      </property>
     </widget>
     <widget class="QTableWidget" name="tableWidget">
      <property name="geometry">
       <rect>
        <x>790</x>
        <y>10</y>
        <width>671</width>
        <height>271</height>
       </rect>
      </property>
     </widget>
     <widget class="QGraphicsView" name="graphicsView">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>300</y>
        <width>1431</width>
        <height>541</height>
       </rect>
      </property>
     </widget>
     <widget class="QSlider" name="zoomSlider">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>770</y>
        <width>231</width>
        <height>61</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Orientation::Horizontal</enum>
      </property>
     </widget>
     <widget class="QPushButton" name="btnStep">
      <property name="geometry">
       <rect>
        <x>640</x>
        <y>260</y>
        <width>91</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Next Step</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btnReset">
      <property name="geometry">
       <rect>
        <x>509</x>
        <y>260</y>
        <width>91</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Reset</string>
      </property>
     </widget>
     <widget class="QLabel" name="lblTextStats">
      <property name="geometry">
       <rect>
        <x>500</x>
        <y>30</y>
        <width>161</width>
        <height>161</height>
       </rect>
      </property>
      <property name="text">
       <string>Statistics</string>
      </property>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>810</y>
        <width>71</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Zoom In/Out</string>
      </property>
     </widget>
     <widget class="Line" name="line">
      <property name="geometry">
       <rect>
        <x>490</x>
        <y>30</y>
        <width>161</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Orientation::Horizontal</enum>
      </property>
     </widget>
     <widget class="Line" name="line_2">
      <property name="geometry">
       <rect>
        <x>490</x>
        <y>180</y>
        <width>161</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Orientation::Horizontal</enum>
      </property>
     </widget>
     <widget class="Line" name="line_3">
      <property name="geometry">
       <rect>
        <x>480</x>
        <y>40</y>
        <width>20</width>
        <height>151</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Orientation::Vertical</enum>
      </property>
     </widget>
     <widget class="Line" name="line_4">
      <property name="geometry">
       <rect>
        <x>640</x>
        <y>40</y>
        <width>20</width>
        <height>151</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Orientation::Vertical</enum>
      </property>
     </widget>
     <widget class="QPushButton" name="btnSaveTextHuff">
      <property name="geometry">
       <rect>
        <x>680</x>
        <y>110</y>
        <width>91</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Save</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="pageImage">
     <widget class="QPushButton" name="btnBack2">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>850</y>
        <width>80</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Back</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btnSelectImage">
      <property name="geometry">
       <rect>
        <x>330</x>
        <y>50</y>
        <width>121</width>
        <height>41</height>
       </rect>
      </property>
      <property name="text">
       <string>Select Image</string>
      </property>
     </widget>
     <widget class="QLabel" name="labelOriginal">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>110</y>
        <width>281</width>
        <height>261</height>
       </rect>
      </property>
      <property name="text">
       <string>                                   Original Image</string>
      </property>
     </widget>
     <widget class="QLabel" name="labelCompressed">
      <property name="geometry">
       <rect>
        <x>460</x>
        <y>110</y>
        <width>251</width>
        <height>261</height>
       </rect>
      </property>
      <property name="text">
       <string>                           Compressed Image</string>
      </property>
     </widget>
     <widget class="QLabel" name="labelStats">
      <property name="geometry">
       <rect>
        <x>770</x>
        <y>140</y>
        <width>151</width>
        <height>151</height>
       </rect>
      </property>
      <property name="text">
       <string>Statistics</string>
      </property>
     </widget>
     <widget class="QTableWidget" name="tableImage">
      <property name="geometry">
       <rect>
        <x>1000</x>
        <y>10</y>
        <width>511</width>
        <height>331</height>
       </rect>
      </property>
     </widget>
     <widget class="QGraphicsView" name="graphicsViewImage">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>360</y>
        <width>1471</width>
        <height>481</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="btnResetImage">
      <property name="geometry">
       <rect>
        <x>730</x>
        <y>320</y>
        <width>80</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Reset</string>
      </property>
     </widget>
     <widget class="QSlider" name="zoomSliderImage">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>790</y>
        <width>171</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Orientation::Horizontal</enum>
      </property>
     </widget>
     <widget class="QPushButton" name="btnCompressImage">
      <property name="geometry">
       <rect>
        <x>350</x>
        <y>230</y>
        <width>80</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Compress</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btnStepImage">
      <property name="geometry">
       <rect>
        <x>870</x>
        <y>320</y>
        <width>80</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Next Step</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>810</y>
        <width>71</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Zoom In/Out</string>
      </property>
     </widget>
     <widget class="Line" name="line_5">
      <property name="geometry">
       <rect>
        <x>760</x>
        <y>280</y>
        <width>161</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Orientation::Horizontal</enum>
      </property>
     </widget>
     <widget class="Line" name="line_6">
      <property name="geometry">
       <rect>
        <x>910</x>
        <y>140</y>
        <width>20</width>
        <height>151</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Orientation::Vertical</enum>
      </property>
     </widget>
     <widget class="Line" name="line_7">
      <property name="geometry">
       <rect>
        <x>750</x>
        <y>140</y>
        <width>20</width>
        <height>151</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Orientation::Vertical</enum>
      </property>
     </widget>
     <widget class="Line" name="line_8">
      <property name="geometry">
       <rect>
        <x>760</x>
        <y>130</y>
        <width>161</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Orientation::Horizontal</enum>
      </property>
     </widget>
     <widget class="QPushButton" name="btnSaveHuff">
      <property name="geometry">
       <rect>
        <x>350</x>
        <y>320</y>
        <width>80</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Save</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="pageHome">
     <widget class="QLabel" name="Title">
      <property name="geometry">
       <rect>
        <x>240</x>
        <y>50</y>
        <width>361</width>
        <height>91</height>
       </rect>
      </property>
      <property name="text">
       <string>Hachiman Compressor
 An Implementation of Huffman Encoding</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btnTextMode">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>230</y>
        <width>131</width>
        <height>51</height>
       </rect>
      </property>
      <property name="text">
       <string>Text Compression</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btnImageMode">
      <property name="geometry">
       <rect>
        <x>380</x>
        <y>230</y>
        <width>131</width>
        <height>51</height>
       </rect>
      </property>
      <property name="text">
       <string>Image Compression</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btnDecompressTextMode">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>330</y>
        <width>131</width>
        <height>51</height>
       </rect>
      </property>
      <property name="text">
       <string>Text Decompression</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btnDecompressImageMode">
      <property name="geometry">
       <rect>
        <x>380</x>
        <y>330</y>
        <width>131</width>
        <height>51</height>
       </rect>
      </property>
      <property name="text">
       <string>Image Decompression</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="pageTextDecomp">
     <widget class="QTextEdit" name="textInputFreqs">
      <property name="geometry">
       <rect>
        <x>200</x>
        <y>70</y>
        <width>381</width>
        <height>91</height>
       </rect>
      </property>
     </widget>
     <widget class="QTextBrowser" name="textOutputDecomp">
      <property name="geometry">
       <rect>
        <x>200</x>
        <y>180</y>
        <width>371</width>
        <height>192</height>
       </rect>
      </property>
     </widget>
     <widget class="QTextEdit" name="textInputBits">
      <property name="geometry">
       <rect>
        <x>660</x>
        <y>70</y>
        <width>381</width>
        <height>91</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="btnDecompressTextAction">
      <property name="geometry">
       <rect>
        <x>640</x>
        <y>260</y>
        <width>80</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Decompress</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btnBack3">
      <property name="geometry">
       <rect>
        <x>190</x>
        <y>950</y>
        <width>80</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Back</string>
      </property>
     </widget>
     <widget class="QGraphicsView" name="graphicsViewTextDecomp">
      <property name="geometry">
       <rect>
        <x>190</x>
        <y>400</y>
        <width>1431</width>
        <height>541</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>280</x>
        <y>910</y>
        <width>71</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Zoom In/Out</string>
      </property>
     </widget>
     <widget class="QSlider" name="zoomSlider_2">
      <property name="geometry">
       <rect>
        <x>210</x>
        <y>860</y>
        <width>231</width>
        <height>61</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Orientation::Horizontal</enum>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="pageImageDecomp">
     <widget class="QPushButton" name="btnOpenHuff">
      <property name="geometry">
       <rect>
        <x>460</x>
        <y>90</y>
        <width>80</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Open file</string>
      </property>
     </widget>
     <widget class="QLabel" name="labelDecompImage">
      <property name="geometry">
       <rect>
        <x>430</x>
        <y>200</y>
        <width>251</width>
        <height>151</height>
       </rect>
      </property>
      <property name="text">
       <string>Decompressed Image</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btnBack4">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>930</y>
        <width>80</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Back</string>
      </property>
     </widget>
     <widget class="QGraphicsView" name="graphicsViewImageDecomp">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>370</y>
        <width>1431</width>
        <height>541</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>870</y>
        <width>71</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Zoom In/Out</string>
      </property>
     </widget>
     <widget class="QSlider" name="zoomSliderImage_2">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>850</y>
        <width>171</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Orientation::Horizontal</enum>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1569</width>
     <height>19</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
